### ISSUE TITLE
Search scrollback history feature

### ISSUE THREAD

**Issue by avrelaun** [Keruspe/Germinal#15]
I do not know if ZSH can search scrollback history on its own, but I do know that tmux handles it.

Since I'm not using tmux on every germinal instance, could it be possible to implement a germinal scrollback search feature ?

**Comment by Keruspe** [comment]
I consider this feature as the same kind of features as tabs / split / tiling / etc. 
I don’t think it’s the terminal emulator job to do that. A terminal emulator is there to emulate a terminal: e.g. display output from the software it’s connected to.
When using tmux, it takes care of all of this.
When not using tmux, some shells might have options to do that, I guess

**Comment by avrelaun** [comment]
I looked around for pure zsh alternatives, couldn't find one.
I like the idea of the separation of concerns, but I do not think it applies in this case.
Searching through previous commands and inputs are the responsability of the shell and several solutions exists.
But in this case this is a pure character displaying feature; One could see it as a gloryfied scroll, only with a param to go to the next occurrence regardless of whether this is an command input or output. Something a shell cannot do btw.

**Comment by Keruspe** [comment]
I'll do it at some point, I just have a couple of other things to fix beforehand

**Comment by avrelaun** [comment]
Gentle poke to remind about this feature, it would have saved me quite some time lately.

**Comment by annacrombie** [comment]
I agree that this feature is better handled by tmux/screen/etc.  The terminal emulator imo should _only_ emulate a terminal

**Comment by avrelaun** [comment]
I beg to differ, as this feature is quite essential to most terminal usages nowadays, and it effectively couples Germinal with a terminal multiplexer.
